packages/database/src/realtime/Connection.ts

```diff
@@ -337,14 +337,18 @@ export class Connection {
     if (MESSAGE_DATA in controlData) {
       const payload = controlData[MESSAGE_DATA];
       if (cmd === SERVER_HELLO) {
-        this.onHandshake_(
-          payload as {
+        const handshakePayload = {
+          ...(payload as {
             ts: number;
             v: string;
             h: string;
             s: string;
-          }
-        );
+          })
+        };
+        if (this.repoInfo_.isUsingEmulator) {
+          handshakePayload.h = this.repoInfo_.host;
```

Comment:
Overriding the handshake hostname feels like a pretty big hammer, so it's probably worth leaving a comment here explaining that emulators will return "localhost" here and that this will break clients not running on the same machine.

Reply:
Yeah, this is a hack to get around it.
Ideally, we should modify the `hostInfo` in the emulator to return the right port.